### DISCUSSION TITLE
[ FF7 ] Steam FFNx remaps Xbox One Controller buttons strangely

### DISCUSSION THREAD

**Discussion by mdesmedt** [julianxhokaxhiu/FFNx#411] [Q&A] [answered]
**Describe the bug**
I'm using FF7 on Steam with a wireless XBox One controller (the latest you can buy right now, with dpad which almost looks like a solid circle). This controller behaves normally in every other game.

By default, the Steam re-release of FF7 has a [well known slightly strange button layout](https://steamcommunity.com/sharedfiles/filedetails/?id=179295659). And I too experience this mapping:
A = OK
B = Cancel
X = Menu
Y = Switch

However, with the FFNx mod installed (both v1.10.1.0 and 10.1.243) my controllers buttons suddenly map to:
A = Cancel
B = Menu
X = Ok
Y = Switch

**To Reproduce**

Install FF7 PC on Steam.
Connect an Xbox one controller.
Verify XBox one controller configuration in Steam is disabled (so there's no funny business with remapping inputs).
Play vanilla FF7 and observe controls as expected.
Install the mod.
Play modded FF7 and observe controls are now different and weird.

**Expected behavior**
I expect either the default FF7 PC Steam button layout, or the PS1 style button layout. I definitely do not expect X to be mapped to OK.

**Comment by julianxhokaxhiu** [comment, marked as answer]
Hello @mdesmedt this is not a bug but a misconfiguration in your `ff7input.cfg` file that is being given by Steam. If you delete that file you'll get the default FF7 map which is the one FFNx uses. Feel free after that moment to reconfigure your inputs inside the game itself.

**Reply by mdesmedt** [reply]
I deleted this file and started the game. But I can still reproduce the issue with the mod installed. Somehow FFNx changes my button layout in the way I described in the bug. Should I reopen it?
Edit: Or is there something else I can try? My FF7 install is totally standard other than trying FFNx. I have not previously tried messing with the button layout.

**Reply by julianxhokaxhiu** [reply]
No need to open a bug, I moved it here on discussions because it is not. How are you running the game? Are you using FFNx in vanilla mode?

**Reply by mdesmedt** [reply]
I'm running the game by clicking Play through Steam, which launches the launcher and clicking Play from there.

By the way I didn't realize you could reconfigure the inputs inside the game. By remapping it inside the game I am able to get a usable controls scheme. Thanks! It's debatable whether this behavior qualifies as a bug I guess :)

**Comment by mdesmedt** [comment]
![20220305143929_1](https://user-images.githubusercontent.com/73319561/156902062-706f31fd-086b-4322-b610-1fffc0967ede.jpg)
This is the gamepad layout I've had to configure to get "normal" FF7 controls with my Xbox controller. The default settings (without ff7input.cfg, with the mod) were quite different.

**Reply by julianxhokaxhiu** [reply]
Yes, because when you click Play through FF7 Steam launcher the `ff7input.cfg` is generated again and you inherit steams layout. Unfortunately can't do much about it, sorry. FFNx is not meant to be invasive, it reads only files, it doesn't write them. The ONLY exception is the flag `save_textures` which generates PNG exports of game textures to allow modders to mod them.

Glad you figured your way out. Enjoy!

**Reply by mdesmedt** [reply]
Ah. Well it makes sense now. Thanks. I would still suggest adding a note about this to the faq.md, basically reminding users what you told me here: Controls might be strange due to ff7input.cfg, just remap them ingame. Thanks for the mod, I look forward to playing through FF7 again 👍

**Reply by julianxhokaxhiu** [reply]
I'll consider adding this into the FAQ, thanks :)
